docs/testplan/DWRR-Snappi-Test-Plan.md

```diff
@@ -0,0 +1,797 @@
+        P2L1[Logical Port 2.1 100G]
+        P2L2[Logical Port 2.2 100G]
+        P2L8[Logical Port 2.8 100G]
+        
+        P3L1[Logical Port 3.1100G]
+        P3L2[Logical Port 3.2100G]
+        P3L8[Logical Port 3.8100G]
+        
+        P4L1[Logical Port 4.1100G]
+        P4L2[Logical Port 4.2100G]
+        P4L8[Logical Port 4.8100G]
+    end
+    
+    style P1L8 fill:#fff,stroke-dasharray: 5 5
+    style P2L8 fill:#fff,stroke-dasharray: 5 5
+    style P3L8 fill:#fff,stroke-dasharray: 5 5
+    style P4L8 fill:#fff,stroke-dasharray: 5 5
+```
+
+*For all intents and purposes, the `Traffic Analyzer` (in the diagrams) is the ingress traffic generator port.*
+
+### 1.3 Equipment Requirements
+
+- **Device Under Test (DUT)**: 
+  - Network switch with DWRR scheduling capabilities
+- **Traffic Generator**: 
+  - Test equipment with snappi API support (Ixia, Keysight, etc.)
+  - 4 physical ports (800G each) with 8 logical ports (100G each) per physical port
+
+### 1.4 Pre-Test Requirements
+
+1. The switch must be configured with DWRR scheduling policies
+2. DSCP to queue mappings must be appropriately configured
+3. Traffic generator must be properly connected to the switch
+4. All interfaces must be in the "up" state
+
+### 1.5 General Test Methodology
+
+For each test scenario:
+
+1. Query the switch for existing DSCP-to-queue mappings and scheduler configurations
+2. Configure traffic flows with appropriate DSCP values based on switch configuration
+3. Generate traffic that causes congestion at the egress port
+4. Capture traffic at the egress port
+5. Analyze the distribution of traffic based on DSCP values
+6. Validate that the observed distribution matches the expected distribution based on DWRR weights (within 2% tolerance)
+
+## 2. Test Scenarios
+
+### 2.1 Scenario 1: Basic DWRR Validation with 2 Queues
+
+#### 2.1.1 Test Objective
+Validate DWRR functionality with 2 queues using standard weight ratios (60:40).
+
+#### 2.1.2 Topology
+
+```mermaid
+flowchart LR
+    subgraph "Traffic Generator"
+        TX1[TX Port 1 100GDSCP 160G]
+        TX2[TX Port 2 100GDSCP 260G]
+    end
+
+    subgraph "Switch Under Test"
+        SW[DUT]
+        Q1[Queue 1 Weight: 60%]
+        Q2[Queue 2 Weight: 40%]
+    end
+
+    subgraph "Traffic Analyzer"
+        RX[RX Port  100G]
+    end
+
+    TX1 -->|DSCP 1| SW
+    TX2 -->|DSCP 2| SW
+    SW --> Q1
+    SW --> Q2
+    Q1 -->|Expected 60%| RX
+    Q2 -->|Expected 40%| RX
+
+    classDef congested fill:#f96,stroke:#333,stroke-width:2px;
+    class RX congested;
+```
+
+#### 2.1.3 Test Configuration
+- **Ingress Ports**: 2 logical ports (100G each)
+- **Egress Port**: 1 logical port (100G)
+- **Traffic Pattern**:
+  - Port 1: 60G of traffic with DSCP value 1 (mapped to Queue 1)
+  - Port 2: 60G of traffic with DSCP value 2 (mapped to Queue 2)
+  - Total ingress traffic: 120G (creates 20% congestion on 100G egress port)
+- **Traffic Flow Duration**: 60 seconds
+- **Packet Size**: 1500 bytes
```

Comment:
should we run the test using different packet size? I believe we will be interested in 8K packet (typical AI) or small packets (stress the pipeline)